pkg/compactor/block_upload.go

```diff
@@ -559,3 +676,49 @@ func (c *MultitenantCompactor) loadValidation(ctx context.Context, userBkt objst
 
 	return v, nil
 }
+
+func (c *MultitenantCompactor) uploadValidationWithError(ctx context.Context, blockID ulid.ULID,
+	userBkt objstore.Bucket, errorStr string) error {
+	val := validationFile{
+		LastUpdate: time.Now().UnixMilli(),
+		Error:      errorStr,
+	}
+	dst := path.Join(blockID.String(), validationFilename)
+	if err := marshalAndUploadToBucket(ctx, userBkt, dst, val); err != nil {
+		return errors.Wrapf(err, "failed uploading %s to bucket", validationFilename)
+	}
+	return nil
+}
+
+func (c *MultitenantCompactor) uploadValidation(ctx context.Context, logger log.Logger, blockID ulid.ULID,
+	userBkt objstore.Bucket) error {
+	return c.uploadValidationWithError(ctx, blockID, userBkt, "")
+}
+
+func (c *MultitenantCompactor) periodicValidationUpdater(ctx context.Context, logger log.Logger, blockID ulid.ULID,
+	userBkt objstore.Bucket, ch chan string, wg *sync.WaitGroup, cancelFn func()) {
+	defer wg.Done()
+	for {
+		select {
+		case <-ch:
+			return
+		case <-time.After(1 * time.Minute):
```

Comment:
nit: `Timer` leak until the duration elapses if the `<-ch` case wins: [docs](https://pkg.go.dev/time#After)